### ISSUE TITLE
Multiple subscriptions for MQTT component are routed to one API endpoint 

### ISSUE THREAD

**Issue by rkrawiel** [dapr/components-contrib#1709]
## Expected Behavior

When having multiple subscriptions to different topics should route them to configured API endpoints.

## Actual Behavior

When having more than one subscription all messages are routed to last registered API endpoint (handler)

## Steps to Reproduce the Problem
Having two subscriptions with configuration like this:
Config
```
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: testpubsub
spec:
  type: pubsub.mqtt
  version: v1
  metadata:
  - name: url
    value: "xxx"
  - name: qos
    value: 1
  - name: retain
    value: "false"
  - name: cleanSession
    value: "true"
  - name: backOffMaxRetries
    value: "5"
```
Subscription 1
```
apiVersion: dapr.io/v1alpha1
kind: Subscription
metadata:
  name: subscription1 
spec:
  pubsubname: testpubsub
  topic: topic1
  route: /api/route1
```
Subscription 2
```
apiVersion: dapr.io/v1alpha1
kind: Subscription
metadata:
  name: subscription2
spec:
  pubsubname: testpubsub
  topic: topic2
  route: /api/route2
```
With this configuration messages from both topics will be routed to /api/route2 which is not expected behavior. 

**Initial investigation leads to mqtt.go Subscribe function where it's registering all topics to one handler (line 216)**

**Comment by rkrawiel** [comment]
@yaron2  Unfortunately, I don't have enough knowledge of the codebase and Go lang itself.

**Comment by ItalyPaleAle** [comment]
Fixed by #1737
